pkg/operation/botanist/operatingsystemconfig.go

```diff
@@ -19,6 +19,7 @@ import (
 	"fmt"
 
 	"github.com/gardener/gardener/charts"
+	"github.com/gardener/gardener/extensions/pkg/controller/worker"
```

Comment:
AFAIK, we should avoid importing packages from github.com/gardener/gardener/extensions into files outside of `./extensions`. I guess this is also the reason to provide this folder during gardenlet build. 

I think you can move the current implementation of `WorkerPoolHash` in https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/worker.go and then use it from `github.com/gardener/gardener/extensions/pkg/controller/worker.WorkerPoolHash`.

Reply:
Thanks for the suggestion! I wasn't really sure where this should live and utils seems like a good place.